src/Pact/Types/ChainId.hs

```diff
@@ -32,12 +37,33 @@ import Pact.Types.Pretty
 -- | Expresses unique platform-specific chain identifier.
 --
 newtype ChainId = ChainId { _chainId :: Text }
-  deriving (Eq, Generic, IsString, ToJSON, FromJSON, Serialize)
-  deriving newtype (NFData)
+  deriving stock (Eq, Generic)
+  deriving newtype
+    ( Show, Pretty, IsString
+    , ToJSON, FromJSON
+    , Serialize, ToTerm, NFData
+    )
 
-instance Show ChainId where show (ChainId c) = show c
-instance Pretty ChainId where pretty = viaShow
+instance Wrapped ChainId
 
-instance ToTerm ChainId where toTerm (ChainId i) = toTerm i
+-- | Lens into the text value of a 'ChainId'
+--
+chainId :: Lens' ChainId Text
```

Comment:
We have `Wrapped`, do we need these? Don't forget there's the `wrap`/`unwrap`/`rewrap` stuff https://github.com/kadena-io/pact/blob/c5dfc9fa2871bcacf90c61a23ecfbbd2040b0637/src/Pact/Types/Util.hs#L194

Reply:
Sometimes it's nice to use the more discoverable optics corresponding with the accessors. Chainweb has these and they've proven useful. Instead of using `Wrapped` or the isos in `Util`, we could also use `view coerced`!